cannot use imported stacks and constructs from libs/packages in nx monorepo

There is an issue when importing stacks/constructs in an aws-cdk infra app created using @ago-dev/nx-aws-cdk-v2. 

This was a surprise issue for me given that one of the key reasons to use an nx workspace is to be able to split up projects into different libs/packages and import them into project apps. This approach is recommended by nx's authors nrwl.io as well.

I am not sure however I wonder if this may be related to not using something like `tsconfig-paths/register` or another solution (tsc-alias etc?) to resolve paths in the definition of `nodeCommandWithRelativePath` in `nx-aws-cdk-v2/packages/aws-cdk-v2/src/utils/executor.util.ts`.

If the issue is indeed related to paths, an unrelated project that might be of interest to reference is nx-builders by Web & Söhne that registers TS paths in its solution for supporting ts-node in the monorepo: https://github.com/tailoredmedia/backend-nx-skeleton/tree/master/packages/nx-builders#ts-node-dev. 

Also of potential interest: https://github.com/aws/jsii/issues/865

Replication:
- create a new nx node:lib or js:lib and add a stack or construct to it
- import the stack and/or construct in the aws-cdk infra app created using the generator from @ago-dev/nx-aws-cdk-v2 and rig it up 
- try to deploy the stack
- observe the error "Cannot find module @myorg/my-library" where "@myorg/my-library" is the `importPath` of the package/library within the nx workspace

Example error output from trying to deploy:

```txt
OptionsParsedExecutorInterface {"_":["TestBucketStack"],"parseArgs":{"_":["TestBucketStack"]},"sourceRoot":"apps/infra/src","root":"apps/infra"}
nodeCommandWithRelativePath node /home/username/projects/my-nx-project/node_modules/aws-cdk/bin/cdk.js deploy
Executing command: node /home/username/projects/my-nx-project/node_modules/aws-cdk/bin/cdk.js deploy --_ TestBucketStack
Error: Cannot find module '@myorg/cdk-stacks'
Require stack:
- /home/username/projects/my-nx-project/apps/infra/src/main.ts
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:995:15)
    at Function.Module._resolveFilename.sharedData.moduleResolveFilenameHook.installedValue [as _resolveFilename] (/home/username/projects/my-nx-project/node_modules/.pnpm/@cspotcode+source-map-support@0.8.1/node_modules/@cspotcode/source-map-support/source-map-support.js:811:30)
    at Function.Module._load (node:internal/modules/cjs/loader:841:27)
    at Module.require (node:internal/modules/cjs/loader:1061:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (/home/username/projects/my-nx-project/apps/infra/src/main.ts:2:1)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module.m._compile (/home/username/projects/my-nx-project/node_modules/.pnpm/ts-node@10.9.1_72wjdbglblrsksmmmhzpvueijy/node_modules/ts-node/src/index.ts:1618:23)
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Object.require.extensions.<computed> [as .ts] (/home/username/projects/my-nx-project/node_modules/.pnpm/ts-node@10.9.1_72wjdbglblrsksmmmhzpvueijy/node_modules/ts-node/src/index.ts:1621:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/home/username/projects/my-nx-project/apps/infra/src/main.ts' ]
}
```

Versions:
- running current versions of everything including nx 15.8.6 (latest) + @ago-dev/nx-aws-cdk-v2 1.3.1

If there's a way to successfully import constants / functions / constructs + stacks from other libraries within the nx workspace, please let me know! Thanks for your work on this package!

--- a/packages/aws-cdk-v2/src/executors/bootstrap/bootstrap.spec.ts
+++ b/packages/aws-cdk-v2/src/executors/bootstrap/bootstrap.spec.ts
@@ -1,7 +1,7 @@
 import * as childProcess from 'child_process';
 import * as path from 'path';
 
-import { logger } from '@nrwl/devkit';
+import { detectPackageManager, logger } from '@nrwl/devkit';
 
 import { BootstrapExecutorSchema } from './schema';
 import executor from './bootstrap';
@@ -9,6 +9,15 @@ import { LARGE_BUFFER } from '../../utils/executor.util';
 import { mockExecutorContext } from '../../utils/testing';
 
 const options: BootstrapExecutorSchema = {};
+
+const NX_WORKSPACE_ROOT = process.env.NX_WORKSPACE_ROOT ?? '';
+if (!NX_WORKSPACE_ROOT) {
+  throw new Error('CDK not Found');
+}
+
+const packageManager = detectPackageManager();
+const generatePath = `"${packageManager} dlx ts-node --require tsconfig-paths/register --project ${NX_WORKSPACE_ROOT}/tsconfig.base.json"`;
+const nodeCommandWithRelativePath = `node --require ts-node/register ${NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js -a ${generatePath} bootstrap`;
 
 describe('aws-cdk-v2 Bootstrap Executor', () => {
   const context = mockExecutorContext('bootstrap');
@@ -24,7 +33,7 @@ describe('aws-cdk-v2 Bootstrap Executor', () => {
     await executor(options, context);
 
     expect(childProcess.exec).toHaveBeenCalledWith(
-      `node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js bootstrap`,
+      nodeCommandWithRelativePath,
       expect.objectContaining({
         cwd: expect.stringContaining(path.join(context.root, context.workspace.projects['proj'].root)),
         env: process.env,
@@ -32,9 +41,7 @@ describe('aws-cdk-v2 Bootstrap Executor', () => {
       })
     );
 
-    expect(logger.debug).toHaveBeenLastCalledWith(
-      `Executing command: node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js bootstrap`
-    );
+    expect(logger.debug).toHaveBeenLastCalledWith(`Executing command: ${nodeCommandWithRelativePath}`);
   });
 
   it('run cdk bootstrap command profile', async () => {
@@ -44,7 +51,7 @@ describe('aws-cdk-v2 Bootstrap Executor', () => {
     await executor(option, context);
 
     expect(childProcess.exec).toHaveBeenCalledWith(
-      `node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js bootstrap --profile ${profile}`,
+      `${nodeCommandWithRelativePath} --profile ${profile}`,
       expect.objectContaining({
         env: process.env,
         maxBuffer: LARGE_BUFFER,
@@ -52,7 +59,7 @@ describe('aws-cdk-v2 Bootstrap Executor', () => {
     );
 
     expect(logger.debug).toHaveBeenLastCalledWith(
-      `Executing command: node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js bootstrap --profile ${profile}`
+      `Executing command: ${nodeCommandWithRelativePath} --profile ${profile}`
     );
   });
 });

--- a/packages/aws-cdk-v2/src/executors/bootstrap/bootstrap.spec.ts
+++ b/packages/aws-cdk-v2/src/executors/bootstrap/bootstrap.spec.ts
@@ -5,12 +5,12 @@ import { logger } from '@nrwl/devkit';
 
 import { BootstrapExecutorSchema } from './schema';
 import executor from './bootstrap';
-import { generatePath, LARGE_BUFFER } from '../../utils/executor.util';
+import { generateCommandString, LARGE_BUFFER } from '../../utils/executor.util';
 import { mockExecutorContext } from '../../utils/testing';
 
 const options: BootstrapExecutorSchema = {};
 
-const nodeCommandWithRelativePath = generatePath('bootstrap');
+const nodeCommandWithRelativePath = generateCommandString('bootstrap', 'apps/proj');
 
 describe('aws-cdk-v2 Bootstrap Executor', () => {
   const context = mockExecutorContext('bootstrap');
@@ -28,7 +28,7 @@ describe('aws-cdk-v2 Bootstrap Executor', () => {
     expect(childProcess.exec).toHaveBeenCalledWith(
       nodeCommandWithRelativePath,
       expect.objectContaining({
-        cwd: expect.stringContaining(path.join(context.root, context.workspace.projects['proj'].root)),
+        cwd: expect.stringContaining(context.root),
         env: process.env,
         maxBuffer: LARGE_BUFFER,
       })

--- a/packages/aws-cdk-v2/src/executors/bootstrap/bootstrap.spec.ts
+++ b/packages/aws-cdk-v2/src/executors/bootstrap/bootstrap.spec.ts
@@ -1,23 +1,16 @@
 import * as childProcess from 'child_process';
 import * as path from 'path';
 
-import { detectPackageManager, logger } from '@nrwl/devkit';
+import { logger } from '@nrwl/devkit';
 
 import { BootstrapExecutorSchema } from './schema';
 import executor from './bootstrap';
-import { LARGE_BUFFER } from '../../utils/executor.util';
+import { generatePath, LARGE_BUFFER } from '../../utils/executor.util';
 import { mockExecutorContext } from '../../utils/testing';
 
 const options: BootstrapExecutorSchema = {};
 
-const NX_WORKSPACE_ROOT = process.env.NX_WORKSPACE_ROOT ?? '';
-if (!NX_WORKSPACE_ROOT) {
-  throw new Error('CDK not Found');
-}
-
-const packageManager = detectPackageManager();
-const generatePath = `"${packageManager} dlx ts-node --require tsconfig-paths/register --project ${NX_WORKSPACE_ROOT}/tsconfig.base.json"`;
-const nodeCommandWithRelativePath = `node --require ts-node/register ${NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js -a ${generatePath} bootstrap`;
+const nodeCommandWithRelativePath = generatePath('bootstrap');
 
 describe('aws-cdk-v2 Bootstrap Executor', () => {
   const context = mockExecutorContext('bootstrap');

--- a/packages/aws-cdk-v2/src/executors/bootstrap/bootstrap.ts
+++ b/packages/aws-cdk-v2/src/executors/bootstrap/bootstrap.ts
@@ -1,5 +1,3 @@
-import * as path from 'path';
-
 import { BootstrapExecutorSchema } from './schema';
 import { createCommand, parseArgs, runCommandProcess } from '../../utils/executor.util';
 import { ParsedExecutorInterface } from '../../interfaces/parsed-executor.interface';
@@ -23,7 +21,7 @@ export default async function runExecutor(options: BootstrapExecutorSchema, cont
 
 function runBootstrap(options: ParsedBootstrapExecutorOption, context: ExecutorContext) {
   const command = createCommand('bootstrap', options);
-  return runCommandProcess(command, path.join(context.root, options.root));
+  return runCommandProcess(command, context.root);
 }
 
 function normalizeOptions(options: BootstrapExecutorSchema, context: ExecutorContext): ParsedBootstrapExecutorOption {

--- a/packages/aws-cdk-v2/src/executors/deploy/deploy.spec.ts
+++ b/packages/aws-cdk-v2/src/executors/deploy/deploy.spec.ts
@@ -4,12 +4,12 @@ import { logger } from '@nrwl/devkit';
 
 import executor from './deploy';
 import { DeployExecutorSchema } from './schema';
-import { generatePath, LARGE_BUFFER } from '../../utils/executor.util';
+import { generateCommandString, LARGE_BUFFER } from '../../utils/executor.util';
 import { mockExecutorContext } from '../../utils/testing';
 
 const options: DeployExecutorSchema = {};
 
-const nodeCommandWithRelativePath = generatePath('deploy');
+const nodeCommandWithRelativePath = generateCommandString('deploy', 'apps/proj');
 
 describe('aws-cdk-v2 deploy Executor', () => {
   const context = mockExecutorContext('deploy');
@@ -27,7 +27,7 @@ describe('aws-cdk-v2 deploy Executor', () => {
     expect(childProcess.exec).toHaveBeenCalledWith(
       nodeCommandWithRelativePath,
       expect.objectContaining({
-        cwd: expect.stringContaining(path.join(context.root, context.workspace.projects['proj'].root)),
+        cwd: context.root,
         env: process.env,
         maxBuffer: LARGE_BUFFER,
       })

--- a/packages/aws-cdk-v2/src/executors/deploy/deploy.spec.ts
+++ b/packages/aws-cdk-v2/src/executors/deploy/deploy.spec.ts
@@ -1,22 +1,15 @@
 import * as path from 'path';
 import * as childProcess from 'child_process';
-import { detectPackageManager, logger } from '@nrwl/devkit';
+import { logger } from '@nrwl/devkit';
 
 import executor from './deploy';
 import { DeployExecutorSchema } from './schema';
-import { LARGE_BUFFER } from '../../utils/executor.util';
+import { generatePath, LARGE_BUFFER } from '../../utils/executor.util';
 import { mockExecutorContext } from '../../utils/testing';
 
 const options: DeployExecutorSchema = {};
 
-const NX_WORKSPACE_ROOT = process.env.NX_WORKSPACE_ROOT ?? '';
-if (!NX_WORKSPACE_ROOT) {
-  throw new Error('CDK not Found');
-}
-
-const packageManager = detectPackageManager();
-const generatePath = `"${packageManager} dlx ts-node --require tsconfig-paths/register --project ${NX_WORKSPACE_ROOT}/tsconfig.base.json"`;
-const nodeCommandWithRelativePath = `node --require ts-node/register ${NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js -a ${generatePath} deploy`;
+const nodeCommandWithRelativePath = generatePath('deploy');
 
 describe('aws-cdk-v2 deploy Executor', () => {
   const context = mockExecutorContext('deploy');

--- a/packages/aws-cdk-v2/src/executors/deploy/deploy.spec.ts
+++ b/packages/aws-cdk-v2/src/executors/deploy/deploy.spec.ts
@@ -1,6 +1,6 @@
 import * as path from 'path';
 import * as childProcess from 'child_process';
-import { logger } from '@nrwl/devkit';
+import { detectPackageManager, logger } from '@nrwl/devkit';
 
 import executor from './deploy';
 import { DeployExecutorSchema } from './schema';
@@ -8,6 +8,15 @@ import { LARGE_BUFFER } from '../../utils/executor.util';
 import { mockExecutorContext } from '../../utils/testing';
 
 const options: DeployExecutorSchema = {};
+
+const NX_WORKSPACE_ROOT = process.env.NX_WORKSPACE_ROOT ?? '';
+if (!NX_WORKSPACE_ROOT) {
+  throw new Error('CDK not Found');
+}
+
+const packageManager = detectPackageManager();
+const generatePath = `"${packageManager} dlx ts-node --require tsconfig-paths/register --project ${NX_WORKSPACE_ROOT}/tsconfig.base.json"`;
+const nodeCommandWithRelativePath = `node --require ts-node/register ${NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js -a ${generatePath} deploy`;
 
 describe('aws-cdk-v2 deploy Executor', () => {
   const context = mockExecutorContext('deploy');
@@ -23,16 +32,14 @@ describe('aws-cdk-v2 deploy Executor', () => {
     await executor(options, context);
 
     expect(childProcess.exec).toHaveBeenCalledWith(
-      `node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js deploy`,
+      nodeCommandWithRelativePath,
       expect.objectContaining({
         cwd: expect.stringContaining(path.join(context.root, context.workspace.projects['proj'].root)),
         env: process.env,
         maxBuffer: LARGE_BUFFER,
       })
     );
-    expect(logger.debug).toHaveBeenLastCalledWith(
-      `Executing command: node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js deploy`
-    );
+    expect(logger.debug).toHaveBeenLastCalledWith(`Executing command: ${nodeCommandWithRelativePath}`);
   });
 
   it('run cdk deploy command stack', async () => {
@@ -42,16 +49,14 @@ describe('aws-cdk-v2 deploy Executor', () => {
     await executor(option, context);
 
     expect(childProcess.exec).toHaveBeenCalledWith(
-      `node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js deploy ${stackName}`,
+      `${nodeCommandWithRelativePath} ${stackName}`,
       expect.objectContaining({
         env: process.env,
         maxBuffer: LARGE_BUFFER,
       })
     );
 
-    expect(logger.debug).toHaveBeenLastCalledWith(
-      `Executing command: node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js deploy ${stackName}`
-    );
+    expect(logger.debug).toHaveBeenLastCalledWith(`Executing command: ${nodeCommandWithRelativePath} ${stackName}`);
   });
 
   it('run cdk deploy command context options', async () => {
@@ -61,7 +66,7 @@ describe('aws-cdk-v2 deploy Executor', () => {
     await executor(option, context);
 
     expect(childProcess.exec).toHaveBeenCalledWith(
-      `node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js deploy --context ${contextOptionString}`,
+      `${nodeCommandWithRelativePath} --context ${contextOptionString}`,
       expect.objectContaining({
         env: process.env,
         maxBuffer: LARGE_BUFFER,
@@ -69,7 +74,7 @@ describe('aws-cdk-v2 deploy Executor', () => {
     );
 
     expect(logger.debug).toHaveBeenLastCalledWith(
-      `Executing command: node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js deploy --context ${contextOptionString}`
+      `Executing command: ${nodeCommandWithRelativePath} --context ${contextOptionString}`
     );
   });
 
@@ -81,16 +86,14 @@ describe('aws-cdk-v2 deploy Executor', () => {
 
     const contextCmd = contextOptions.map((option) => `--context ${option}`).join(' ');
     expect(childProcess.exec).toHaveBeenCalledWith(
-      `node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js deploy ${contextCmd}`,
+      `${nodeCommandWithRelativePath} ${contextCmd}`,
       expect.objectContaining({
         env: process.env,
         maxBuffer: LARGE_BUFFER,
       })
     );
 
-    expect(logger.debug).toHaveBeenLastCalledWith(
-      `Executing command: node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js deploy ${contextCmd}`
-    );
+    expect(logger.debug).toHaveBeenLastCalledWith(`Executing command: ${nodeCommandWithRelativePath} ${contextCmd}`);
   });
 
   it('run cdk deploy command with boolean context option', async () => {
@@ -99,15 +102,13 @@ describe('aws-cdk-v2 deploy Executor', () => {
     await executor(option, context);
 
     expect(childProcess.exec).toHaveBeenCalledWith(
-      `node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js deploy --context true`,
+      `${nodeCommandWithRelativePath} --context true`,
       expect.objectContaining({
         env: process.env,
         maxBuffer: LARGE_BUFFER,
       })
     );
 
-    expect(logger.debug).toHaveBeenLastCalledWith(
-      `Executing command: node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js deploy --context true`
-    );
+    expect(logger.debug).toHaveBeenLastCalledWith(`Executing command: ${nodeCommandWithRelativePath} --context true`);
   });
 });

--- a/packages/aws-cdk-v2/src/executors/deploy/deploy.ts
+++ b/packages/aws-cdk-v2/src/executors/deploy/deploy.ts
@@ -22,7 +22,7 @@ export default async function runExecutor(options: DeployExecutorSchema, context
 
 async function runDeploy(options: ParsedDeployExecutorOption, context: ExecutorContext) {
   const command = createCommand('deploy', options);
-  return runCommandProcess(command, path.join(context.root, options.root));
+  return runCommandProcess(command, context.root);
 }
 
 function normalizeOptions(options: DeployExecutorSchema, context: ExecutorContext): ParsedDeployExecutorOption {

--- a/packages/aws-cdk-v2/src/executors/destroy/destroy.spec.ts
+++ b/packages/aws-cdk-v2/src/executors/destroy/destroy.spec.ts
@@ -5,12 +5,12 @@ import { logger } from '@nrwl/devkit';
 
 import { DestroyExecutorSchema } from './schema';
 import executor from './destroy';
-import { generatePath, LARGE_BUFFER } from '../../utils/executor.util';
+import { generateCommandString, LARGE_BUFFER } from '../../utils/executor.util';
 import { mockExecutorContext } from '../../utils/testing';
 
 const options: DestroyExecutorSchema = {};
 
-const nodeCommandWithRelativePath = generatePath('destroy');
+const nodeCommandWithRelativePath = generateCommandString('destroy', 'apps/proj');
 
 describe('aws-cdk-v2 Destroy Executor', () => {
   const context = mockExecutorContext('destroy');
@@ -28,7 +28,7 @@ describe('aws-cdk-v2 Destroy Executor', () => {
     expect(childProcess.exec).toHaveBeenCalledWith(
       nodeCommandWithRelativePath,
       expect.objectContaining({
-        cwd: expect.stringContaining(path.join(context.root, context.workspace.projects['proj'].root)),
+        cwd: context.root,
         env: process.env,
         maxBuffer: LARGE_BUFFER,
       })

--- a/packages/aws-cdk-v2/src/executors/destroy/destroy.spec.ts
+++ b/packages/aws-cdk-v2/src/executors/destroy/destroy.spec.ts
@@ -1,23 +1,16 @@
 import * as childProcess from 'child_process';
 import * as path from 'path';
 
-import { detectPackageManager, logger } from '@nrwl/devkit';
+import { logger } from '@nrwl/devkit';
 
 import { DestroyExecutorSchema } from './schema';
 import executor from './destroy';
-import { LARGE_BUFFER } from '../../utils/executor.util';
+import { generatePath, LARGE_BUFFER } from '../../utils/executor.util';
 import { mockExecutorContext } from '../../utils/testing';
 
 const options: DestroyExecutorSchema = {};
 
-const NX_WORKSPACE_ROOT = process.env.NX_WORKSPACE_ROOT ?? '';
-if (!NX_WORKSPACE_ROOT) {
-  throw new Error('CDK not Found');
-}
-
-const packageManager = detectPackageManager();
-const generatePath = `"${packageManager} dlx ts-node --require tsconfig-paths/register --project ${NX_WORKSPACE_ROOT}/tsconfig.base.json"`;
-const nodeCommandWithRelativePath = `node --require ts-node/register ${NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js -a ${generatePath} destroy`;
+const nodeCommandWithRelativePath = generatePath('destroy');
 
 describe('aws-cdk-v2 Destroy Executor', () => {
   const context = mockExecutorContext('destroy');

--- a/packages/aws-cdk-v2/src/executors/destroy/destroy.spec.ts
+++ b/packages/aws-cdk-v2/src/executors/destroy/destroy.spec.ts
@@ -1,7 +1,7 @@
 import * as childProcess from 'child_process';
 import * as path from 'path';
 
-import { logger } from '@nrwl/devkit';
+import { detectPackageManager, logger } from '@nrwl/devkit';
 
 import { DestroyExecutorSchema } from './schema';
 import executor from './destroy';
@@ -9,6 +9,15 @@ import { LARGE_BUFFER } from '../../utils/executor.util';
 import { mockExecutorContext } from '../../utils/testing';
 
 const options: DestroyExecutorSchema = {};
+
+const NX_WORKSPACE_ROOT = process.env.NX_WORKSPACE_ROOT ?? '';
+if (!NX_WORKSPACE_ROOT) {
+  throw new Error('CDK not Found');
+}
+
+const packageManager = detectPackageManager();
+const generatePath = `"${packageManager} dlx ts-node --require tsconfig-paths/register --project ${NX_WORKSPACE_ROOT}/tsconfig.base.json"`;
+const nodeCommandWithRelativePath = `node --require ts-node/register ${NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js -a ${generatePath} destroy`;
 
 describe('aws-cdk-v2 Destroy Executor', () => {
   const context = mockExecutorContext('destroy');
@@ -24,7 +33,7 @@ describe('aws-cdk-v2 Destroy Executor', () => {
     await executor(options, context);
 
     expect(childProcess.exec).toHaveBeenCalledWith(
-      `node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js destroy`,
+      nodeCommandWithRelativePath,
       expect.objectContaining({
         cwd: expect.stringContaining(path.join(context.root, context.workspace.projects['proj'].root)),
         env: process.env,
@@ -32,9 +41,7 @@ describe('aws-cdk-v2 Destroy Executor', () => {
       })
     );
 
-    expect(logger.debug).toHaveBeenLastCalledWith(
-      `Executing command: node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js destroy`
-    );
+    expect(logger.debug).toHaveBeenLastCalledWith(`Executing command: ${nodeCommandWithRelativePath}`);
   });
 
   it('run cdk destroy command stack', async () => {
@@ -45,15 +52,13 @@ describe('aws-cdk-v2 Destroy Executor', () => {
     await executor(option, context);
 
     expect(childProcess.exec).toHaveBeenCalledWith(
-      `node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js destroy ${stackName}`,
+      `${nodeCommandWithRelativePath} ${stackName}`,
       expect.objectContaining({
         env: process.env,
         maxBuffer: LARGE_BUFFER,
       })
     );
 
-    expect(logger.debug).toHaveBeenLastCalledWith(
-      `Executing command: node ${process.env.NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js destroy ${stackName}`
-    );
+    expect(logger.debug).toHaveBeenLastCalledWith(`Executing command: ${nodeCommandWithRelativePath} ${stackName}`);
   });
 });

--- a/packages/aws-cdk-v2/src/executors/destroy/destroy.ts
+++ b/packages/aws-cdk-v2/src/executors/destroy/destroy.ts
@@ -23,7 +23,7 @@ export default async function runExecutor(options: DestroyExecutorSchema, contex
 
 function runDestroy(options: ParsedDestroyExecutorOption, context: ExecutorContext) {
   const command = createCommand('destroy', options);
-  return runCommandProcess(command, path.join(context.root, options.root));
+  return runCommandProcess(command, context.root);
 }
 
 function normalizeOptions(options: DestroyExecutorSchema, context: ExecutorContext): ParsedDestroyExecutorOption {

--- a/packages/aws-cdk-v2/src/utils/executor.util.ts
+++ b/packages/aws-cdk-v2/src/utils/executor.util.ts
@@ -8,6 +8,18 @@ import { BootstrapExecutorSchema } from '../executors/bootstrap/schema';
 export const executorPropKeys = ['stacks'];
 export const LARGE_BUFFER = 1024 * 1000000;
 
+const NX_WORKSPACE_ROOT = process.env.NX_WORKSPACE_ROOT ?? '';
+if (!NX_WORKSPACE_ROOT) {
+  throw new Error('CDK not Found');
+}
+
+export function generatePath(command: string) {
+  const packageManager = detectPackageManager();
+  const packageManagerExecutor = packageManager === 'npm' ? 'npx' : `${packageManager} dlx`;
+  const generatePath = `"${packageManagerExecutor} ts-node --require tsconfig-paths/register --project ${NX_WORKSPACE_ROOT}/tsconfig.base.json"`;
+  return `node --require ts-node/register ${NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js -a ${generatePath} ${command}`;
+}
+
 export function parseArgs(options: DeployExecutorSchema | BootstrapExecutorSchema): Record<string, string | string[]> {
   const keys = Object.keys(options);
   return keys
@@ -17,14 +29,8 @@ export function parseArgs(options: DeployExecutorSchema | BootstrapExecutorSchem
 
 export function createCommand(command: string, options: ParsedExecutorInterface): string {
   console.log('OptionsParsedExecutorInterface', JSON.stringify(options));
-  const NX_WORKSPACE_ROOT = process.env.NX_WORKSPACE_ROOT ?? '';
-  if (!NX_WORKSPACE_ROOT) {
-    throw new Error('CDK not Found');
-  }
 
-  const packageManager = detectPackageManager();
-  const generatePath = `"${packageManager} dlx ts-node --require tsconfig-paths/register --project ${NX_WORKSPACE_ROOT}/tsconfig.base.json"`;
-  const nodeCommandWithRelativePath = `node --require ts-node/register ${NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js -a ${generatePath} ${command}`;
+  const nodeCommandWithRelativePath = generatePath(command);
   console.log('nodeCommandWithRelativePath', nodeCommandWithRelativePath);
   const commands = [nodeCommandWithRelativePath];
 

--- a/packages/aws-cdk-v2/src/utils/executor.util.ts
+++ b/packages/aws-cdk-v2/src/utils/executor.util.ts
@@ -2,7 +2,7 @@ import { exec } from 'child_process';
 
 import { DeployExecutorSchema } from '../executors/deploy/schema';
 import { ParsedExecutorInterface } from '../interfaces/parsed-executor.interface';
-import { logger } from '@nrwl/devkit';
+import { logger, detectPackageManager } from '@nrwl/devkit';
 import { BootstrapExecutorSchema } from '../executors/bootstrap/schema';
 
 export const executorPropKeys = ['stacks'];
@@ -21,7 +21,10 @@ export function createCommand(command: string, options: ParsedExecutorInterface)
   if (!NX_WORKSPACE_ROOT) {
     throw new Error('CDK not Found');
   }
-  const nodeCommandWithRelativePath = `node ${NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js ${command}`;
+
+  const packageManager = detectPackageManager();
+  const generatePath = `"${packageManager} dlx ts-node --require tsconfig-paths/register --project ${NX_WORKSPACE_ROOT}/tsconfig.base.json"`;
+  const nodeCommandWithRelativePath = `node --require ts-node/register ${NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js -a ${generatePath} ${command}`;
   console.log('nodeCommandWithRelativePath', nodeCommandWithRelativePath);
   const commands = [nodeCommandWithRelativePath];
 

--- a/packages/aws-cdk-v2/src/utils/executor.util.ts
+++ b/packages/aws-cdk-v2/src/utils/executor.util.ts
@@ -13,10 +13,11 @@ if (!NX_WORKSPACE_ROOT) {
   throw new Error('CDK not Found');
 }
 
-export function generatePath(command: string) {
+export function generateCommandString(command: string, appPath: string) {
   const packageManager = detectPackageManager();
   const packageManagerExecutor = packageManager === 'npm' ? 'npx' : `${packageManager} dlx`;
-  const generatePath = `"${packageManagerExecutor} ts-node --require tsconfig-paths/register --project ${NX_WORKSPACE_ROOT}/tsconfig.base.json"`;
+  const projectPath = `${NX_WORKSPACE_ROOT}/${appPath}`;
+  const generatePath = `"${packageManagerExecutor} ts-node --require tsconfig-paths/register --project ${projectPath}/tsconfig.app.json ${projectPath}/src/main.ts"`;
   return `node --require ts-node/register ${NX_WORKSPACE_ROOT}/node_modules/aws-cdk/bin/cdk.js -a ${generatePath} ${command}`;
 }
 
@@ -30,7 +31,7 @@ export function parseArgs(options: DeployExecutorSchema | BootstrapExecutorSchem
 export function createCommand(command: string, options: ParsedExecutorInterface): string {
   console.log('OptionsParsedExecutorInterface', JSON.stringify(options));
 
-  const nodeCommandWithRelativePath = generatePath(command);
+  const nodeCommandWithRelativePath = generateCommandString(command, options.root);
   console.log('nodeCommandWithRelativePath', nodeCommandWithRelativePath);
   const commands = [nodeCommandWithRelativePath];
 

